src/pages/index.js

```diff
@@ -0,0 +1,307 @@
+
+const deleteModalCancelButton = deleteModal.querySelector(".modal__submit-button_cancel");
+deleteModalCancelButton.addEventListener("click", () => {
+  closeModal(deleteModal);
+});
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true, "Create", "Creating...");
+
+  const inputValues = {
+    name: cardModalCaptionInput.value,
+    link: cardModalLinkInput.value
+  };
+
+  api
+    .createCard(inputValues)
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      cardFormElement.reset();
+      closeModal(cardModal);
+    })
+    .catch((error) => {
+      console.error(error);
+    })
+    .finally(() => {
+      setButtonText(submitButton, false, "Create", "Creating...");
+      const buttonElement = cardFormElement.querySelector(settings.submitButtonSelector);
+      disableButton(buttonElement, settings);
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = avatarFormElement.querySelector(".modal__submit-button");
+  setButtonText(submitButton, true, "Save", "Saving...");
+
+  api
+    .editAvatarInfo({ avatar: avatarInput.value })
+    .then((data) => {
+      const avatarImage = document.querySelector("#avatar");
+        avatarImage.src = data.avatar;
+        closeModal(avatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(avatarSubmitButton, false);
+    });
+};
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content.querySelector(".card").cloneNode(true);
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikeButton = cardElement.querySelector(".card__like-button");
+  const cardDeleteButton = cardElement.querySelector(".card__delete-button");
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+  cardElement.dataset.cardId = data._id;
+
+  if (data.isLiked) {
+    cardLikeButton.classList.add("card__like-button_liked");
+  }
+
+  cardLikeButton.addEventListener("click", (evt) =>
+    handleLike(evt, data._id, cardLikeButton));
+
+  cardDeleteButton.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id, cardDeleteButton));
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+    previewModalCaptionEl.textContent = data.name;
+  });
+
+  return cardElement;
+}
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  deleteModal.dataset.cardId = cardId;
+  openModal(deleteModal);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  
+  const deleteButton = deleteModal.querySelector('.modal__submit-button');
```

Comment:
Please, don't search the submit button here because in the other submit handlers, you have ` const submitButton = evt.submitter;`


```js

  const submitButton = evt.submitter;

 
```